src/Dictionaries/getDictionaryConfigurationFromAST.cpp

```diff
@@ -609,6 +609,12 @@ getDictionaryConfigurationFromAST(const ASTCreateQuery & query, ContextPtr conte
 
     bool complex = DictionaryFactory::instance().isComplex(dictionary_layout->layout_type);
 
+    if (pk_attrs.size() > 1 && !complex
```

Comment:
Also, we need to use it if the key is not represented as a number up to 64bit. For example, for a single String field, we need to convert to complex as well.

Reply:
https://github.com/ClickHouse/ClickHouse/blob/8509f1b64537e75a91018c1e4948fe6d7cf5c1b1/src/Dictionaries/DictionaryStructure.cpp#L134
Now simple key `id` is always UInt64 regardless of the type of create sql, so we need only converting it to complex without type UInt8~UInt64, right? @alexey-milovidov 

And I find -1 is saved uncorrectly with simple layout and int `id`, but we can create successfully and select with no exception. If converting it to complex, it will be fixed. 
```
CREATE DICTIONARY test.dict_hashed_simple_int128_1
(
    `v0` Int128,
    `v1` UInt16,
    `v2` UInt16
)
PRIMARY KEY v0
SOURCE(CLICKHOUSE(TABLE 'dict_data_int'))
LIFETIME(MIN 0 MAX 0)
LAYOUT(HASHED());

h-199 :) select * from dict_data_int where v0=-1;

SELECT *
FROM dict_data_int
WHERE v0 = -1

Query id: a0fd25b1-251e-4fb8-85f0-a2ae302fc4fe

┌─v0─┬─v1─┬─v2─┬─v3─┬─v4─┐
│ -1 │  1 │  1 │  1 │  1 │
└────┴────┴────┴────┴────┘

1 rows in set. Elapsed: 0.011 sec. 

h-199 :) select * from dict_hashed_simple_int128_1 where v0=-1;

SELECT *
FROM dict_hashed_simple_int128_1
WHERE v0 = -1

Query id: 81f54008-47c8-4d6c-b73d-cdaabbee0e31

Ok.

0 rows in set. Elapsed: 0.004 sec. 

h-199 :) select * from dict_hashed_simple_int128_1 where v1=1;

SELECT *
FROM dict_hashed_simple_int128_1
WHERE v1 = 1

Query id: 947a9441-f1f9-463d-9d04-4a15fcbb4703

┌───────────────────v0─┬─v1─┬─v2─┐
│ 18446744073709551615 │  1 │  1 │
└──────────────────────┴────┴────┘

1 rows in set. Elapsed: 0.005 sec.
```

Reply:
> we need only converting it to complex without type UInt8~UInt64, right?

Yes, let's do it this way.